# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1308582676 14400
#      Mon Jun 20 11:11:16 2011 -0400
# Branch stable
# Node ID ecd1364a270074527060d953aa0f9cf6f7ccaa51
# Parent  824c60c8f7b9edeeb75bc8e4591b9cfc9fa6483a
allow assignment of [], "", and '' to sparse logical matrix

* op-sbm-bm.cc (DEFNULLASSIGNOP_FN (null_assign, ...)):
New assignment operator function.
(install_sbm_bm_ops): Install it for various RHS types.

diff --git a/src/OPERATORS/op-sbm-bm.cc b/src/OPERATORS/op-sbm-bm.cc
--- a/src/OPERATORS/op-sbm-bm.cc
+++ b/src/OPERATORS/op-sbm-bm.cc
@@ -38,16 +38,17 @@ along with Octave; see the file COPYING.
 #include "ov-uint8.h"
 #include "ov-uint16.h"
 #include "ov-uint32.h"
 #include "ov-uint64.h"
 #include "ov-range.h"
 #include "ov-scalar.h"
 #include "ov-str-mat.h"
 #include "ops.h"
+#include "ov-null-mat.h"
 #include "ov-re-sparse.h"
 #include "ov-bool-sparse.h"
 #include "smx-bm-sbm.h"
 #include "smx-sbm-bm.h"
 
 // sparse bool matrix by bool matrix ops.
 
 DEFBINOP_FN (eq, sparse_bool_matrix, bool_matrix, mx_el_eq)
@@ -83,16 +84,18 @@ DEFCATOP (sm_bm, sparse_matrix, bool_mat
 DEFASSIGNOP (assign, sparse_bool_matrix, bool_matrix)
 {
   CAST_BINOP_ARGS (octave_sparse_bool_matrix&, const octave_bool_matrix&);
 
   v1.assign (idx, SparseBoolMatrix (v2.bool_matrix_value ()));
   return octave_value ();
 }
 
+DEFNULLASSIGNOP_FN (null_assign, sparse_bool_matrix, delete_elements)
+
 static octave_value
 oct_assignop_conv_and_assign (octave_base_value& a1,
                               const octave_value_list& idx,
                               const octave_base_value& a2)
 {
   octave_sparse_bool_matrix& v1 = dynamic_cast<octave_sparse_bool_matrix&> (a1);
 
   // FIXME -- perhaps add a warning for this conversion if the values
@@ -149,9 +152,16 @@ install_sbm_bm_ops (void)
   INSTALL_ASSIGNOP (op_asn_eq, octave_sparse_bool_matrix, octave_uint8_matrix,
                     conv_and_assign);
   INSTALL_ASSIGNOP (op_asn_eq, octave_sparse_bool_matrix, octave_uint16_matrix,
                     conv_and_assign);
   INSTALL_ASSIGNOP (op_asn_eq, octave_sparse_bool_matrix, octave_uint32_matrix,
                     conv_and_assign);
   INSTALL_ASSIGNOP (op_asn_eq, octave_sparse_bool_matrix, octave_uint64_matrix,
                     conv_and_assign);
+
+  INSTALL_ASSIGNOP (op_asn_eq, octave_sparse_bool_matrix, octave_null_matrix,
+                    null_assign);
+  INSTALL_ASSIGNOP (op_asn_eq, octave_sparse_bool_matrix, octave_null_str,
+                    null_assign);
+  INSTALL_ASSIGNOP (op_asn_eq, octave_sparse_bool_matrix, octave_null_sq_str,
+                    null_assign);
 }
